crates/common/fork_choice/src/store.rs

```diff
@@ -801,6 +801,24 @@ impl Store {
 
         Ok(())
     }
+
+    pub fn is_syncing(&self) -> anyhow::Result<bool> {
+        let head = self.get_head()?;
+
+        let head_slot = match self.db.beacon_block_provider().get(head) {
+            Ok(Some(block)) => block.message.slot,
+            err => {
+                return Err(anyhow!("Failed to get head slot, error: {err:?}"));
+            }
+        };
+
+        // calculate sync_distance
+        let current_slot = self.get_current_slot()?;
+
+        let sync_distance = current_slot.saturating_sub(head_slot);
```

Comment:
```suggestion
        let head_slot = match self.db.beacon_block_provider().get(self.get_head()?) {
            Ok(Some(block)) => block.message.slot,
            err => {
                return Err(anyhow!("Failed to get head slot, error: {err:?}"));
            }
        };

        // calculate sync_distance
        let sync_distance = self.get_current_slot()?.saturating_sub(head_slot);
```

we might as well inline these right values as making variables for them doesn't add much value